docs/vscode.md

```diff
@@ -358,6 +381,30 @@ echo "Hello world!"
 """
 ```
 
+**Type conversions**
+
+Groovy supports two ways to perform type conversions:
+
+```groovy
+def map = (Map) readJson(json)  // soft cast
+def map = readJson(json) as Map // hard cast
+```
+
+The Nextflow language specification only supports hard casts. However, hard casts are discouraged because they can cause unexpected behavior if used improperly. You can use a Groovy-style type annotation instead:
```

Comment:
```suggestion
The Nextflow language specification only supports hard casts. However, hard casts are discouraged because they can cause unexpected behavior if used improperly. Use a Groovy-style type annotation instead:
```